jupyterhub/tests/test_utils.py

```diff
@@ -100,6 +100,23 @@ async def test_tornado_coroutines():
     assert (await t.tornado_coroutine()) == "gen.coroutine"
 
 
+@pytest.mark.parametrize(
+    "pieces, expected",
+    [
+        (("/"), "/"),
+        (("/", "/"), "/"),
+        (("/base", ""), "/base"),
+        (("/base/", ""), "/base/"),
+        (("/base", "abc", "def"), "/base/abc/def"),
+        (("/base/", "/abc/", "/def/"), "/base/abc/def/"),
+        ((""), ""),
```

Comment:
Some more test cases to cover:

```suggestion
        (("/base", "", "/", ""), "/base/"),
        ((""), ""),
        (("", ""), ""),
        (("", "part", ""), "part"),
        (("", "/part"), "part"),
        (("", "part", "", "after"), "part/after"),
        (("", "part", "", "after/", "", ""), "part/after/"),
```